### PR TITLE
Add Gutunkunst OOA QC

### LINKED ISSUE #492: HomSap/OutOfAfrica_3G09 needs QCing
Someone this model has not yet been QCed!

### PR DESCRIPTION
**Pull request by apragsdale**
Implemented qc version of the 2009 OOA model (to close #492). I'm having issues getting the tests to catch if I made a mistake in implementation, and I think the development docs are out of date (specifically, item 5 under Demographic model review process, perhaps)? Do I need to add anything beyond `qc_model = homo_sapiens_qc.GutenkunstOOA()` to the `TestGutenkunstThreePopOutOfAfrica` class in `test_homo_sapiens.py`?

### CODE REVIEW

File: tests/test_homo_sapiens.py [hunk 1]
```diff
@@ -109,11 +109,13 @@ class TestJacobsPapuansOutOfAfrica(
     qc_model = homo_sapiens_qc.DenisovanAncestryInPapuans()
 
 
-# Models that have not been QC'd:
-
 class TestGutenkunstThreePopOutOfAfrica(
         unittest.TestCase, test_models.CatalogDemographicModelTestMixin):
     model = species.get_demographic_model("OutOfAfrica_3G09")
```
Comment (grahamgower) [inline tests/test_homo_sapiens.py:114]: Same issue here as for the ZigZag model, regarding the need to instead inherit from `QcdCatalogDemographicModelTestMixin`.

**Comment by grahamgower** [comment]
Thanks for taking on these missing QC models @apragsdale! Should be a simple change to get them doing the checks. And when #304 and #305 are resolved, no one else will get caught on changing which superclass to inherit from!

**Comment by apragsdale** [comment]
Thanks, I had forgotten that needed to be changed. Coming back to implement models a few months later really does make it clear where the docs could be filled in :) I can do some updating of the QC process documentation after these are finished up.

**Comment by apragsdale** [comment]
I did find a mistake in the OOA model currently in the catalog: in demographic events, migration was not turned off between the African and Bottleneck populations after they are merged, so that migration between those populations continues into the distant past.

I don't know if this will have affected any of the analyses in the paper, for example Figure 3 from the manuscript... But if so, we should probably discuss on the call tomorrow.

**Comment by grahamgower** [comment]
That's a little worrying! @jeromekelleher can you comment on the likely behaviour of msprime here? Does a `migration_matrix` with non-zero entries have any effect for lineages already "joined" by a MassMigration(proportion=1) event?

**Comment by apragsdale** [comment]
It definitely will, because migration will carry lineages back into the merged population, so you'll end up with longer expected Tmrcas for lineages still uncoalesced beyond that merger event. Since it's in the distant past, the effect will likely be small, but it will still be there.

**Comment by jeromekelleher** [comment]
> That's a little worrying! @jeromekelleher can you comment on the likely behaviour of msprime here? Does a migration_matrix with non-zero entries have any effect for lineages already "joined" by a MassMigration(proportion=1) event?

We don't touch the migration matrix when there's MassMigrations - decided events having side effects was confusing.

@apragsdale, does this mean that the OOA model that's in the [msprime docs](https://msprime.readthedocs.io/en/latest/tutorial.html#id1) is wrong too? (Ouch!)

**Comment by apragsdale** [comment]
> @apragsdale, does this mean that the OOA model that's in the [msprime docs](https://msprime.readthedocs.io/en/latest/tutorial.html#id1) is wrong too? (Ouch!)

Oh no!! It certainly looks like it.. That's surprising that hasn't been caught before, really. I've used that example in the msprime docs a bunch of times and never noticed it either.

**Comment by apragsdale** [comment]
I think this (and the Zigzag model) are ready to merge. Let me know if you need me to rebase at all again for one or both.

**Comment by petrelharp** [comment]
I have no idea why Travis is unhappy here - it is something to do with conda. Help, @grahamgower? @jradrion?

**Comment by grahamgower** [comment]
It's an unrelated problem, and it looks like its been failing for two weeks now. Travis is running fine for linux on python 3.6. The failing builds are for osx --- maybe somethign changed at the travis end, and the `.travis.yml` file needs to have a language set explicitly for the osx case?

**Comment by jeromekelleher** [comment]
Travis build should be fixed now if you rebase @apragsdale.

**Comment by apragsdale** [comment]
This and #498 are now both rebased.

**Comment by petrelharp** [comment]
Yay, thanks!!!

**Pull request merged**